src/client/components/DiceInterface.tsx

```diff
@@ -160,62 +388,90 @@ export function DiceInterface() {
-                      onClick={() => addDiceType("d20")}
+                      onClick={() => {
+                        addDiceType("d20");
+                        focusIndexFromRef(d20Ref);
+                      }}
+                      ref={d20Ref} // diceTypeRefs[diceTypeRefs.length - 1]
                     >
                       d20
                     </Button>
                     <Button
                       style={{ width: "30%", fontSize: "0.7rem" }}
                       className="green"
                       onClick={() => addDiceType("a20")}
+                      ref={d20AdvRef}
                     >
                       ADV
                     </Button>
                     <Button
                       style={{ width: "30%", fontSize: "0.7rem" }}
                       className="red"
                       onClick={() => addDiceType("i20")}
+                      ref={d20DisRef}
                     >
                       DIS
                     </Button>
                   </div>
                 </td>
                 <td className="flex flex-wrap flex-col h-[182px]">
-                  {[-1, -2, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((bonus) => (
-                    <Button
-                      className="h-1/6 w-1/2 flex-1"
-                      key={bonus}
-                      onClick={() => addBonus(bonus)}
-                    >
-                      {bonus > 0 ? "+" + bonus.toString() : bonus}
-                    </Button>
-                  ))}
+                  {modTypes.map((bonus) => {
+                    const ref = modRefs[modTypes.indexOf(bonus)];
+                    return (
+                      <Button
+                        className="h-1/6 w-1/2 flex-1"
+                        key={bonus}
+                        onClick={() => {
+                          addBonus(bonus);
+                          focusIndexFromRef(ref);
+                        }}
+                        ref={ref}
+                      >
+                        {bonus > 0 ? "+" + bonus.toString() : bonus}
+                      </Button>
+                    );
+                  })}
                 </td>
 
                 <td>
                   <Button
                     className="w-full"
                     disabled={!character}
-                    onClick={() => doRoll(true)}
+                    onClick={async () => {
+                      await doRoll(true);
+                      focusIndexFromRef(tempRef);
+                    }}
+                    ref={tempRef}
                     title={character ? undefined : "No character selected"}
                   >
                     <p>Template</p>
                   </Button>
                   <Button
                     className="w-full h-[120px]"
-                    onClick={() => doRoll(false)}
+                    onClick={async () => {
+                      await doRoll(false);
+                      focusIndexFromRef(rollRef);
```

Comment:
Maybe swap these two calls around so that the index is updated before the asynchronous call to `doRoll`?